js/views/transactions/Tab.js

```diff
@@ -0,0 +1,217 @@
+import $ from 'jquery';
+import app from '../../app';
+import baseVw from '../baseVw';
+import loadTemplate from '../../utils/loadTemplate';
+import TransactionsTable from './table/Table';
+
+export default class extends baseVw {
+  constructor(options = {}) {
+    const opts = {
+      defaultFilter: {
+        search: '',
+        sort: 'date-desc',
+        state: [2, 3, 4, 5, 6, 7, 8, 9, 10],
+      },
+      ...options,
+    };
+
+    super(opts);
+
+    if (!this.collection) {
+      throw new Error('Please provide a transactions collection.');
+    }
+
+    const types = ['sales', 'purchases', 'cases'];
+
+    if (types.indexOf(opts.type) === -1) {
+      throw new Error(`Type needs to be one of ${types}.`);
+    }
+
+    if (!opts.filterConfig) {
+      throw new Error('Please provide a filter config object.');
+    }
+
+    this.options = opts || {};
+    this.type = opts.type;
+    this.filterConfig = opts.filterConfig;
+    this.acceptPosts = {};
+    this.rejectPosts = {};
+    this.cancelPosts = {};
+    this.filter = { ...opts.defaultFilter };
+
+    this.listenTo(this.collection, 'request', (cl, xhr) => {
+      setTimeout(() => {
+        if (this.table && this.table.curPage === 1) {
+          this.$queryTotalLine.empty();
+
+          xhr.done(data => {
+            const count =
+              `<span class="txB">
+                ${app.polyglot.t(`transactions.${this.type}.countTransactions`,
+                  { smart_count: data.queryCount })}
+               </span>`;
+            this.$queryTotalLine.html(
+              app.polyglot.t(`transactions.${this.type}.countTransactionsFound`,
+                { countTransactions: count })
+            );
+          });
+        }
+      });
+    });
+  }
+
+  className() {
+    return `${this.type} tx5`;
+  }
+
+  events() {
+    return {
+      'change .filter input': 'onChangeFilter',
+      'keyup .js-searchInput': 'onKeyUpSearch',
+    };
+  }
+
+  onChangeFilter() {
+    let state = [];
+    this.filter.state = this.$filterCheckboxes.filter(':checked')
```

Comment:
good catch. this went through a refactor or two, so that was probably lost in the shuffle.